core/src/main/java/jenkins/tasks/SimpleBuildStep.java

```diff
@@ -80,9 +82,38 @@
      * @param listener a place to send output
      * @throws InterruptedException if the step is interrupted
      * @throws IOException if something goes wrong; use {@link AbortException} for a polite error
+     * @deprecated Use {@link #perform(Run, FilePath, EnvVars, Launcher, TaskListener) instead.}
      */
-    void perform(@NonNull Run<?,?> run, @NonNull FilePath workspace, @NonNull Launcher launcher,
-                 @NonNull TaskListener listener) throws InterruptedException, IOException;
+    @Deprecated
+    default void perform(@NonNull Run<?, ?> run, @NonNull FilePath workspace, @NonNull Launcher launcher,
+                         @NonNull TaskListener listener) throws InterruptedException, IOException {
+        // No additional environment available; just use that from the Run.
+        this.perform(run, workspace, run.getEnvironment(listener), launcher, listener);
+    }
+
+    /**
+     * Run this step.
+     * @param run a build this is running as a part of
+     * @param workspace a workspace to use for any file operations
+     * @param env environment variables applicable to this step
+     * @param launcher a way to start processes
+     * @param listener a place to send output
+     * @throws InterruptedException if the step is interrupted
+     * @throws IOException if something goes wrong; use {@link AbortException} for a polite error
+     * @since 2.2xx
```

Comment:
```suggestion
     * @since FIXME
```

(gets patched up by a core committer after merge)